tests/components/alexa/test_smart_home.py

```diff
@@ -0,0 +1,207 @@
+            assert appliance['friendlyName'] == "Test light 1"
+            assert appliance['actions'] == ['turnOff', 'turnOn']
+            continue
+
+        if appliance['applianceId'] == 'light#test_2':
+            assert appliance['applianceTypes'][0] == "LIGHT"
+            assert appliance['friendlyName'] == "Test light 2"
+            assert appliance['actions'] == \
+                ['turnOff', 'turnOn', 'setPercentage']
+            continue
+
+        raise AssertionError("Unknown appliance!")
+
+
+@asyncio.coroutine
+def test_api_entity_not_exists(hass):
+    """Test api turn on process without entity."""
+    msg_switch = smart_home.api_message(
+        'TurnOnRequest', 'Alexa.ConnectedHome.Control', {
+            'appliance': {
+                'applianceId': 'switch#test'
+            }
+        })
+
+    call_switch = async_mock_service(hass, 'switch', 'turn_on')
+
+    resp = yield from smart_home.async_api_turn_on(hass, msg_switch)
+    assert len(call_switch) == 0
+    assert resp['header']['name'] == 'DriverInternalError'
+    assert resp['header']['namespace'] == 'Alexa.ConnectedHome.Control'
+
+
+@asyncio.coroutine
+def test_api_turn_on(hass):
+    """Test api turn on process."""
+    msg_switch = smart_home.api_message(
+        'TurnOnRequest', 'Alexa.ConnectedHome.Control', {
+            'appliance': {
+                'applianceId': 'switch#test'
+            }
+        })
+    msg_light = smart_home.api_message(
+        'TurnOnRequest', 'Alexa.ConnectedHome.Control', {
+            'appliance': {
+                'applianceId': 'light#test'
+            }
+        })
+
+    # settup test devices
+    hass.states.async_set(
+        'switch.test', 'off', {'friendly_name': "Test switch"})
+
+    hass.states.async_set(
+        'light.test', 'off', {'friendly_name': "Test light"})
+
+    call_switch = async_mock_service(hass, 'switch', 'turn_on')
+    call_light = async_mock_service(hass, 'light', 'turn_on')
+
+    resp = yield from smart_home.async_api_turn_on(hass, msg_switch)
+    assert len(call_switch) == 1
+    assert call_switch[0].data['entity_id'] == 'switch.test'
+    assert resp['header']['name'] == 'TurnOnConfirmation'
+
+    resp = yield from smart_home.async_api_turn_on(hass, msg_light)
+    assert len(call_light) == 1
+    assert call_light[0].data['entity_id'] == 'light.test'
+    assert resp['header']['name'] == 'TurnOnConfirmation'
+
+
+@asyncio.coroutine
+def test_api_turn_off(hass):
+    """Test api turn off process."""
+    msg_switch = smart_home.api_message(
+        'TurnOffRequest', 'Alexa.ConnectedHome.Control', {
+            'appliance': {
+                'applianceId': 'switch#test'
+            }
+        })
+    msg_light = smart_home.api_message(
+        'TurnOffRequest', 'Alexa.ConnectedHome.Control', {
+            'appliance': {
+                'applianceId': 'light#test'
+            }
+        })
+
+    # settup test devices
```

Comment:
No need to create mock states since you're mocking the services

Reply:
We check and need the state object inside api